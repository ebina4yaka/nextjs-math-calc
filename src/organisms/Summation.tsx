import React, { useState } from 'react'
import SummationFormula from '../molecules/SummationFormula'
import SummationInputField, {
  CalcSummation,
} from '../molecules/SummationInputField'

export default function Summation(): React.ReactElement {
  const [n, setN] = useState(10)
  const [k, setK] = useState(1)
  const [a, setA] = useState(1)
  const [answer, setAnswer] = useState(CalcSummation(n, k, a))
  return (
    <>
      <SummationFormula n={n} k={k} a={a} answer={answer} />
      <SummationInputField
        n={n}
        k={k}
        a={a}
        setN={setN}
        setK={setK}
        setA={setA}
        setAnswer={setAnswer}
      />
    </>
  )
}
