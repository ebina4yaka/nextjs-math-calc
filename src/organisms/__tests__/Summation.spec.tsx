import React from 'react'
import renderer from 'react-test-renderer'
import Summation from '../Summation'

describe('Summation Tests', () => {
  test('Snapshot Test', () => {
    const tree = renderer.create(<Summation />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
