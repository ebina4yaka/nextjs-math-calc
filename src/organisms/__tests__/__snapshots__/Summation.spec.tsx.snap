// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summation 1`] = `
Array [
  <div
    className="makeStyles-container-1"
  >
    <div>
      <div
        className="makeStyles-number-3"
      >
        10
      </div>
      <div
        className="makeStyles-sigma-2"
      >
        Î£
      </div>
      <div
        className="makeStyles-number-3"
      >
        k = 
        1
      </div>
    </div>
    <div
      className="makeStyles-number-3"
    >
      1
       * k
    </div>
    <div
      className="makeStyles-answer-4"
    >
      <div
        className="makeStyles-number-3"
      >
         = 
        55
      </div>
    </div>
  </div>,
  <div
    className="makeStyles-container-5"
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-6"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionStart"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            n=
          </p>
        </div>
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value={10}
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-6"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionStart"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            k=
          </p>
        </div>
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value={1}
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-6"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionStart"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            a=
          </p>
        </div>
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value={1}
        />
      </div>
    </div>
  </div>,
]
`;
